server.js

```diff
@@ -119,15 +189,19 @@ function runClient(client) {
     }
 
     answers[data.id][data.number].score++;
-    console.log('we upvoting this shit: ' + answers[data.id][data.number].score);
-    for (var i3 in answers[data.id]) {
-      if (answers[data.id][i3].number == data.number) {
-        socket.emit('upvote', answers[data.id][i3]);
+    log('we upvoting this shit: ' + answers[data.id][data.number].score);
+    for (var z in answers[data.id]) {
+      if (answers[data.id][z].number === data.number) {
+        socket.emit('upvote', answers[data.id][z]);
         break;
       }
     }
     clients[client.id].upvoted.push({id:data.id, number:data.number});
   });
+
+  client.on('error', function(data) {
+    log('Socket Error: ', data);
```

Comment:
Had one case where the server complained about not having an `error` event handler, so I slapped it in here.